### PR TITLE
Set the namespace of the service knative-local-gateway to the istio namespace

### LINKED ISSUE #565: Knative operator looks for "istio-system" namespace when custom istio gateway is set
**Describe the bug**
When we deploy the knative operator and set the custom `knative-ingress-gateway` to another gateway and namespace, the knative serving controller still looks for the `istio-system` namespace and fails to get ready.

operator log: `failed to apply non rbac manifest: namespaces \"istio-system\" not found`

describe of KnativeServing: `Install failed with message: namespaces "istio-system" not found`

**Expected behavior**
When setting the custom gateway, knative doesn't look for the gateway in `istio-system`, but the defined namespace for example `servicemesh`.

**To Reproduce**
Deploy istio in custom namespace and knative with the following configs.

**Knative release version**
Knative Operator: v0.21.0 - v0.22.0

**Additional context**
Serving Configuration:

```---
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeServing
metadata:
  name: platform-knative-serving
spec:
  version: 0.22.0
  config:
    domain:
    istio:
      gateway.knative-serving.knative-ingress-gateway: "app-gateway.servicemesh.svc.cluster.local"
  ingress:
    istio:
      enabled: true
      knative-ingress-gateway:
        selector:
          istio: custom-gateway
```

If we have an obviously misconfiguration please point me out...

### PR DESCRIPTION
**Pull request by houshengbo**
Fixes #565 #586

## Proposed Changes

* Set the namespace of istio to the service knative-local-gateway based on the serving CR.

### CODE REVIEW

File: pkg/reconciler/knativeserving/common/ingress_service.go [hunk 1]
```diff
@@ -17,19 +17,45 @@ limitations under the License.
...
 // It also removes the OwnerReference to the operator, as they are in different namespaces, which is
 // invalid in Kubernetes 1.20+.
-func IngressServiceTransform() mf.Transformer {
+func IngressServiceTransform(ks *v1alpha1.KnativeServing) mf.Transformer {
 	return func(u *unstructured.Unstructured) error {
 		if u.GetAPIVersion() == "v1" && u.GetKind() == "Service" && u.GetName() == "knative-local-gateway" {
 			u.SetNamespace("istio-system")
 			u.SetOwnerReferences(nil)
+			config := ks.GetSpec().GetConfig()
+			if data, ok := config["istio"]; ok {
```
Comment (maximilien) [inline pkg/reconciler/knativeserving/common/ingress_service.go:37]: What if the `config[“istio”]` is not present? Or could that be the case when different ingress is used?

**Review by maximilien — COMMENTED** [review]
/ok-to-test

File: pkg/reconciler/knativeserving/common/ingress_service.go (hunk 1 above)
Comment (houshengbo) [reply pkg/reconciler/knativeserving/common/ingress_service.go:37]: if neither `istio` nor `config-istio` exists, we will use the default ns `u.SetNamespace("istio-system")`

**Review by maximilien — APPROVED** [review]
/approve
/lgtm

**Pull request merged**